### PR TITLE
Makefile: check sse4.2 for tikv targets

### PR DESCRIPTION
**Pull request by overvenus**
## What have you changed? (mandatory)

Add a script to checks whether tikv-server enables sse4.2.

## What are the type of the changes? (mandatory)

- Misc (other changes)

## How has this PR been tested? (mandatory)

CI

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

File: etc/check-sse4_2.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        if [[ -x "$dirfile" && ! -d "$dirfile" ]]; then
+            
+            echo "checking binary $dirfile for sse4.2"
+            
+            # RocksDB uses sse4.2 in the `Fast_CRC32` function
+            fast_crc32=$(nm "$dirfile" | grep -o " _.*Fast_CRC32.*")
+            if [[ ! $fast_crc32 ]]; then
+                echo "error: $dirfile does not contain rocksdb::crc32c::Fast_CRC32 function"
+                errors=1
+                continue
```
Comment (Connor1996) [inline etc/check-sse4_2.sh]: why not break here

Comment (overvenus) [reply etc/check-sse4_2.sh]: Because we want to check other files

Comment (Connor1996) [reply etc/check-sse4_2.sh]: But seems once errors=1, checking other files doesn't matter the result.

Comment (overvenus) [reply etc/check-sse4_2.sh]: Yes, I mean we may want to know how many files are not using sse4.2.

**Comment by breeswish** [comment]
Should we also forbid new instruction sets like AVX and AVX2?

Additionally I think it might not be very good to put this script in TiKV repo because it is not a really TiKV stuff 🤣

**Comment by overvenus** [comment]
> Should we also forbid new instruction sets like AVX and AVX2?

This script only ensures tikv has sse4.2 support, does not care about other instruction sets.

> Additionally I think it might not be very good to put this script in TiKV repo because it is not a really TiKV stuff

Yes, so I place it under the `etc` folder and write comments for CI use only.

**Pull request merged**